### ISSUE TITLE
Dropwizard could not log INFO log to console and log file at very early stage of application initialization

### ISSUE THREAD

**Issue by dknc** [dropwizard/dropwizard#1658]
Dropwizard could not log INFO log to console and log file at very early stage of application initialization. 

My dropwizard application implemented a StrSubstitutor and configured it in the initialize method

``` @Override
  public void initialize(Bootstrap<ChatServiceAppConfiguration> bootstrap)
  {
    // Enable variable substitution with environment variables
    bootstrap.setConfigurationSourceProvider(
        new SubstitutingSourceProvider(
            bootstrap.getConfigurationSourceProvider(),
            new AppPropertySubstitutor(false)
        )
    );
  }
```

The AppPropertySubstitutor depends on a component which uses apache common logging.
All INFO level log is not written to the console and the log file in the depended component.

Have checked that jcl-over-slf4j.jar is successfully packaged into the application jar.

The cause is that the default constructor of the dropwizard Application calls  

```
BootstrapLogging.bootstrap()
```

The default logger initialized a ThresholdFilter with log level Level.WARN, thus INFO log is denied.

As the BootstrapLogging.bootstrap() can only be called once. Super constructor is always executed when creating sub class instance. thus it is not able to bootstrap logging with different log level other than Level.WARN.

Dropwizard should bootstrap logging with the logging configuration defined in yaml at very first stage.

Dropwizard version:  1.0.0-rc4

**Comment by evnm** [comment]
You should be able to get around this by defining a default constructor in your `Application` subclass and invoking `BootstrapLogging.bootstrap(Level)` with the desired log level:

```
import ch.qos.logback.classic.Level;
import io.dropwizard.logging.BootstrapLogging;

public FooApplication extends Application<FooConfiguration> {
  protected FooApplication() {
    BootstrapLogging.bootstrap(Level.DEBUG);
  }

  ...
}
```

**Comment by dknc** [comment]
Override the default constructor of the concrete dropwizard application does not work as super constructor will be always called first and call BootstrapLogging.bootstrap(Level.DEBUG) in sub class make no effect as the bootstrap method will set bootstrapped flag to true after it is called, subsequent call to this method just simply returns.

**Comment by evnm** [comment]
You're right, my mistake. Had to look this one up: https://docs.oracle.com/javase/specs/jls/se8/html/jls-12.html#jls-12.5

[The Javadoc comment on `Application`](https://github.com/dropwizard/dropwizard/blob/master/dropwizard-core/src/main/java/io/dropwizard/Application.java#L15-L17) is misleading, as it insinuates that subclasses _can_ control the level of bootstrapped logging. This should be fixed.

I think it'd be a good idea to provide a means of configuring the log level passed to `BootstrapLogging.bootstrap(Level)`: #1662

**Comment by nickbabcock** [comment]
Closed by #1662 a long time ago
